include/dxc/Support/dxcapi.use.h

```diff
@@ -21,12 +21,14 @@ extern const char *kDxilLib;
 
 // Helper class to dynamically load the dxcompiler or a compatible libraries.
 class DxcDllSupport {
+  friend class DxcDllExtValidationSupport;
```

Comment:
Why is this needed?

(I haven't dug deeply, but `friend` immediately rings alarm bells!)

Reply:
I think you've probably overcomplicated this.  `DxcDllExtValidationSupport` should be using the public interface of `DllSupport` - so it should call `InitializeForDll` to specify which dll/entry point to use.

Really, `friend` exists to support a few very niche cases, and many cases of broken designs.